packages/atomic/package.json

```diff
@@ -13,22 +13,24 @@
   "es2015": "dist/esm/index.mjs",
   "es2017": "dist/esm/index.mjs",
   "types": "dist/types/index.d.ts",
-  "collection": "dist/collection/collection-manifest.json",
-  "collection:main": "dist/collection/index.js",
+  "collection": "collection-manifest.json",
```

Comment:
Stencil is aware that Atomic is a Stencil project, and uses the “collection” output target. The manifest contains the list of Atomic components, which aren’t compiled into Web component. My own Stencil app will then automatically use it’s @stencil/core to compile the Atomic components and lazy load them in my page. This is not compatible with the CDN, and the compilation is invalid as some components don’t render correctly. We should not allow this, Atomic components should only and always be used as Web Component, and not Stencil Components. I haven’t found any specific config to change this, so I wrote my own custom manifest and deleted the generated output in the build to reduce the size. If I just delete the collection/collection:main inside the package.json, the @coveo/atomic import stops working inside my Stencil app.

Reply:
Why should they not be used as stencil components, is it because of the rendering issues?

Reply:
Perhaps we should generate the [dist-custom-elements output](https://stenciljs.com/docs/custom-elements)?
There is [this thread comment](https://github.com/ionic-team/stencil/issues/2531#issuecomment-808082875) on Stencil's Github that might be relevant?

Reply:
There are rendering issues yes, and from our standpoint, I don't see why we would want to output/support Stencil components when it's supposed to simply be a toolkit that builds web component (with our Stencil.config with it's plugins), we would only want to output the resulting web components. From the client's standpoint, they will have a choice of using the CDN or bundling Atomic in their codebase instead of being forced to load and compile our stencil components

Reply:
@samisayegh it won't affect the collection output, different issue. I'm targeting the `dist` instead of the `collection` and it works fine.

That said, we should perhaps have the `dist-custom-elements` output too, Olivier did it here: https://github.com/coveo/ui-kit/pull/1645#discussion_r781329577

Reply:
To target `dist` didn't you have to create an empty `manifest.json` file, adjust the paths in the `package.json`, delete the collection dir etc.? It feels like we're fighting with Stencil's tooling. If we generated `dist-custom-elements`, maybe things would just work?

Reply:
Generating `dist-custom-elements` was one of the first thing I tried (i'll try again to be 100% sure). The Stencil project will just automatically get whats in `collection`. If we don't set collection/collection:main in package.json, the Stencil app won't see @coveo/atomic as a valid module

Reply:
Ok, it sounds like Stencil hasn't considered a stencil project importing another stencil project then,

Reply:
I managed for collection not to be generated instead of deleting it, here was the error when we remove the collection & collection:main from package json
<img width="708" alt="Screen Shot 2022-01-11 at 6 11 12 PM" src="https://user-images.githubusercontent.com/4923043/149161960-310b135c-cb3a-4dd1-bb65-6caec0787999.png">

Reply:
> sounds like Stencil hasn't considered a stencil project importing another stencil project 

They did, it's the purpose of the collection output, but they haven't considered disabling it. We don't want the collection output for the reasons listed above